src/backend/machobj.c

```diff
@@ -2310,6 +2310,7 @@ void MachObj::addrel(int seg, targ_size_t offset, symbol *targsym,
         unsigned targseg, int rtype, int val)
 {
     Relocation rel;
+    memset(&rel, 0, sizeof(rel));
```

Comment:
I don't believe this struct is ever written to disk. It's for use by machobj.c only. Your point about holes is valid. But because this is a performance critical issue for this type, I suggest reorganizing the Relocation struct so that there are no holes. I.e.:
```
struct Relocation
{
    symbol *funcsym;
    symbol *targsym;
    unsigned offset;
    unsigned targseg;
    unsigned rtype;
    int val;
}
```
should do it.